### PR TITLE
fix(picker): update picker display and icon flex properties to resolve layout bugs

### PR DESCRIPTION
**Pull request by cdransf**
## Description

Addresses issue with vertical alignment of picker text/placeholder by applying inline-flex in template rather than inline-block. Applies flex-shrink to validation icons to prevent icon from resizing when label is long enough to be truncated.

CSS-977

## How and where has this been tested?

Verified locally in Storybook.

### Validation steps

1. Fetch branch and run locally or access the Storybook URL for the PR.
2. Navigate to the picker component.
3. Verify that the placeholder text is properly aligned within the picker.
4. Verify that the validation icon does not resize when a long label is supplied as an arg.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

5. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## Screenshots

Screenshot 2024-10-08 at 8 56 30 AM
Screenshot 2024-10-08 at 8 56 35 AM

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/picker/stories/template.js
Comment (jawinn) [inline components/picker/stories/template.js:141]: This inline CSS has been removed in #3200 . CSS-977 was marked as "has to be done after" CSS-936 in Jira, so I would wait until that PR merges.

**Review by jawinn — CHANGES_REQUESTED** [review]

File: components/picker/stories/template.js
Comment (cdransf) [reply components/picker/stories/template.js:141]: Great! I'll keep an eye on that and wait for it to be merged.

**Review by rise-erpelding — APPROVED** [review]
Validated that this works with value and placeholder, with icons, and is not an issue with the loading progress circle!

File: .changeset/chatty-sheep-draw.md
Comment (jawinn) [inline .changeset/chatty-sheep-draw.md:5]:
```suggestion
Addresses issue with vertical alignment of picker text/placeholder by applying inline-flex in template rather than inline-block. Applies flex-shrink to validation icons to prevent icon from resizing when label is long enough to be truncated.
```
The changeset needs updating to reflect the latest changes.

**Review by jawinn — CHANGES_REQUESTED** [review]

File: .changeset/chatty-sheep-draw.md
Comment (cdransf) [reply .changeset/chatty-sheep-draw.md]: Updated to `Applies flex-shrink to validation icons to prevent icon from resizing when label is long enough to be truncated.` to reflect the reduced scope of the changes.

**Review by jawinn — APPROVED** [review]
Looks good, thanks. It just needs the VRTs checked before it merges.

**Pull request merged**
